curator-x-discovery/src/test/java/org/apache/curator/x/discovery/details/NewServiceInstance.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.curator.x.discovery.details;
+
+import com.google.common.base.Preconditions;
+import org.apache.curator.x.discovery.ServiceType;
+import org.apache.curator.x.discovery.UriSpec;
+import org.codehaus.jackson.annotate.JsonTypeInfo;
+import org.codehaus.jackson.annotate.JsonTypeInfo.Id;
+import java.net.URI;
+import java.util.Date;
+
+/**
+ * POJO that represents a service instance
+ */
+class NewServiceInstance<T>
+{
+    private final String name;
+    private final String id;
+    private final String address;
+    private final Integer port;
+    private final Integer sslPort;
+    private final T payload;
+    private final long registrationTimeUTC;
+    private final ServiceType serviceType;
+    private final UriSpec uriSpec;
+    private final boolean enabled;
+    private final String new1;
+    private final Long new2;
+    private final Date new3;
+    private final URI new4;
+
```

Comment:
What are these new1-new4 variables? They seem to be a new addition as part of this fix?

Also, Is there a reason that this class can't be a subclass of the OldServiceInstance?